Detectors/EMCAL/base/include/EMCALBase/EMCGeometry.h

```diff
@@ -0,0 +1,378 @@
+  /// \return center of supermodule in phi
+  ///
+  Double_t GetPhiCenterOfSM(Int_t nsupmod) const;
+  ///
+  /// \return center of supermodule in phi sector
+  ///
+  Double_t GetPhiCenterOfSMSec(Int_t nsupmod) const;
+  Float_t GetSuperModulesPar(Int_t ipar) const { return mParSM[ipar]; }
+  Int_t GetSMType(Int_t nSupMod) const
+  {
+    if (nSupMod > GetNumberOfSuperModules())
+      return NOT_EXISTENT;
+    return mEMCSMSystem[nSupMod];
+  }
+
+  ///
+  /// SM boundaries
+  ///
+  /// \param nSupMod: super module index, input
+  /// \param phiMin: minimum phi value in radians, output
+  /// \param phiMax: maximum phi value in radians, output
+  ///
+  Bool_t GetPhiBoundariesOfSM(Int_t nSupMod, Double_t& phiMin, Double_t& phiMax) const;
+
+  ///
+  /// SM boundaries between gaps
+  ///
+  /// \param nPhiSec: super module sector index, input
+  /// \param phiMin: minimum phi value in radians, output
+  /// \param phiMax: maximum phi value in radians, output
+  ///
+  /// * 0;  gap boundaries between  0th&2th  | 1th&3th SM
+  /// * 1;  gap boundaries between  2th&4th  | 3th&5th SM
+  /// * 2;  gap boundaries between  4th&6th  | 5th&7th SM
+  /// * 3;  gap boundaries between  6th&8th  | 7th&9th SM
+  /// * 4;  gap boundaries between  8th&10th | 9th&11th SM
+  /// * 5;  gap boundaries between 10th&12th | 11h&13th SM
+  ///
+  Bool_t GetPhiBoundariesOfSMGap(Int_t nPhiSec, Double_t& phiMin, Double_t& phiMax) const;
+
+  ///
+  /// Play with strings names and modify them for better handling (?)
+  ///
+  static int ParseString(const TString& topt, TObjArray& Opt);
+
+  ///////////////////////////////
+  // Geometry data member setters
+  //
+  void SetNZ(Int_t nz)
+  {
+    mNZ = nz;
+    LOG(INFO) << "SetNZ: Number of modules in Z set to " << mNZ << FairLogger::endl;
+  }
+  void SetNPhi(Int_t nphi)
+  {
+    mNPhi = nphi;
+    LOG(INFO) << "SetNPhi: Number of modules in Phi set to " << mNPhi << FairLogger::endl;
+  }
+  void SetSampling(Float_t samp)
+  {
+    mSampling = samp;
+    LOG(INFO) << "SetSampling: Sampling factor set to " << mSampling << FairLogger::endl;
+  }
+
+  ///////////////////
+  // useful utilities
+  //
+  /// \return theta in radians for a given pseudorapidity
+  Float_t AngleFromEta(Float_t eta) const { return 2.0 * TMath::ATan(TMath::Exp(-eta)); }
+
+  /// \return z in for a given pseudorapidity and r=sqrt(x*x+y*y).
+  Float_t ZFromEtaR(Float_t r, Float_t eta) const { return r / TMath::Tan(AngleFromEta(eta)); }
+
+  //////////////////////////////////////////////////
+  // Obsolete?
+  Float_t GetSteelFrontThickness() const { return mSteelFrontThick; }
+  //////////////////////////////////////////////////
+
+  static const Char_t* sDefaultGeometryName; ///< Default name of geometry
+  static Bool_t sInit;                       ///< Tells if geometry has been succesfully set up.
+
+ private:
+  // Member data
+
+  TString mGeoName; ///< geometry name
+
+  TObjArray* mArrayOpts;          //!<! array of geometry options
```

Comment:
is it easy to change all `TObjArray` into `std::` containers? Maybe we should do it now.